web/src/main/java/org/cbioportal/web/StudyViewController.java

```diff
@@ -611,6 +611,29 @@ private static double parseValueLinear(ClinicalData c) {
         List<String> studyIds = new ArrayList<>();
         List<String> sampleIds = new ArrayList<>();
         studyViewFilterUtil.extractStudyAndSampleIds(studyViewFilterApplier.apply(interceptedStudyViewFilter), studyIds, sampleIds);
+        List<Sample> filteredSamples = sampleService.fetchSamples(studyIds, sampleIds, Projection.DETAILED.name());
+
+        List<String> studyIdsWithoutNumericalFilter = new ArrayList<>();
+        List<String> sampleIdsWithoutNumericalFilter = new ArrayList<>();
+        
+        if (interceptedStudyViewFilter.getClinicalDataFilters() != null) {
+            // Remove clinical data filter
+            ClinicalDataFilter numericalFilterObj = null;
+            for (ClinicalDataFilter f : interceptedStudyViewFilter.getClinicalDataFilters()) {
+                if (f.getAttributeId().equals(numericalAttributeId)) {
+                    numericalFilterObj = f;
+                    break;
+                }
+            }
+            if (numericalFilterObj != null) {
+                interceptedStudyViewFilter.getClinicalDataFilters().remove(numericalFilterObj);
+            }
```

Comment:
This is looking way down the road, but if we ever upgrade to java 20/21/wherever Loom lands, a stream based find implementation is probably going to work a bit better ootb with virtual threads. That's a big part of why I keep making these sort of suggestions.
```suggestion
            // Remove clinical data filter
            interceptedStudyViewFilter.getClinicalDataFilters().stream()
                .filter(f -> f.getAttributeId().equals(numericalAttributeId))
                .findAny()
                .ifPresent(f -> interceptedStudyViewFilter.getClinicalDataFilters().remove(f));
```